wallet/createtx.go

```diff
@@ -174,27 +174,54 @@ func (w *Wallet) txToOutputs(outputs []*wire.TxOut,
 			return err
 		}
 
-		// Wrap our coins in a type that implements the SelectableCoin
-		// interface, so we can arrange them according to the selected
-		// coin selection strategy.
-		wrappedEligible := make([]Coin, len(eligible))
-		for i := range eligible {
-			wrappedEligible[i] = Coin{
-				TxOut: wire.TxOut{
-					Value:    int64(eligible[i].Amount),
-					PkScript: eligible[i].PkScript,
-				},
-				OutPoint: eligible[i].OutPoint,
+		var inputSource txauthor.InputSource
+		if len(selectedUtxos) > 0 {
+			mapEligibleToOutpoint := make(map[wire.OutPoint]wtxmgr.
+				Credit)
+
+			for _, e := range eligible {
+				mapEligibleToOutpoint[e.OutPoint] = e
+			}
+
+			var eligibleSelectedUtxo []wtxmgr.Credit
+			for _, outpoint := range selectedUtxos {
+				e, ok := mapEligibleToOutpoint[outpoint]
+
+				if !ok {
+					return fmt.Errorf(
```

Comment:
nit: see exception here about `fmt.Errorf()` formatting: https://github.com/lightningnetwork/lnd/blob/master/docs/code_formatting_rules.md#exception-for-log-and-error-message-formatting

So this can just be (there's also a space missing between "outpoint" and "not":
```
					return fmt.Errorf("selected outpoint "+
						"not eligible for spending: "+
						"%v", outpoint)
```